src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -1539,42 +1539,37 @@ namespace winrt::Microsoft::Terminal::Control::implementation
     // - caseSensitive: boolean that represents if the current search is case sensitive
     // Return Value:
     // - <none>
-    void ControlCore::Search(const winrt::hstring& text,
-                             const bool goForward,
-                             const bool caseSensitive)
+    void ControlCore::Search(const winrt::hstring& text, const bool goForward, const bool caseSensitive)
     {
-        if (text.size() == 0)
+        auto lock = _terminal->LockForWriting();
+
+        if (_searcher.IsStale() || _searcherText != text || _searcherGoForward != goForward || _searcherCaseSensitive != caseSensitive)

```

Comment:
hmm. we need some way to tell the searcher that it is stale when we switch into the alt buffer.

Reply:
The alt buffer will not have the same mutation count.

Reply:
There's a one in 900 quin-jillion-millio nchance that it will, but... fair.

Reply:
Yeah, I'll take that chance! 😄 I mean the worst-case scenario is that the search finds results that are stale.